diff --git a/go.mod b/go.mod
index b8ca3bd..fc78009 100644
--- a/go.mod
+++ b/go.mod
@@ -2,11 +2,14 @@ module github.com/hashicorp/go-sockaddr
 
 go 1.24.0
 
+replace github.com/wlynxg/anet => ../anet
+
 require (
 	github.com/hashicorp/errwrap v1.1.0
 	github.com/mitchellh/cli v1.1.5
 	github.com/mitchellh/go-wordwrap v1.0.1
 	github.com/ryanuber/columnize v2.1.2+incompatible
+	github.com/wlynxg/anet v0.0.0
 )
 
 require (
diff --git a/ifaddrs.go b/ifaddrs.go
index 258e124..17b7a13 100644
--- a/ifaddrs.go
+++ b/ifaddrs.go
@@ -13,6 +13,8 @@ import (
 	"sort"
 	"strconv"
 	"strings"
+
+	"github.com/wlynxg/anet"
 )
 
 var (
@@ -245,14 +247,14 @@ func IfAttrs(selectorName string, ifAddrs IfAddrs) (string, error) {
 // available IP addresses on each interface and converts them to
 // sockaddr.IPAddrs, and returning the result as an array of IfAddr.
 func GetAllInterfaces() (IfAddrs, error) {
-	ifs, err := net.Interfaces()
+	ifs, err := anet.Interfaces()
 	if err != nil {
 		return nil, err
 	}
 
 	ifAddrs := make(IfAddrs, 0, len(ifs))
 	for _, intf := range ifs {
-		addrs, err := intf.Addrs()
+		addrs, err := anet.InterfaceAddrsByInterface(&intf)
 		if err != nil {
 			return nil, err
 		}
